### DISCUSSION TITLE
[Meta] Better utilize our prioritization system

### DISCUSSION THREAD

**Discussion by DNin01** [ScratchAddons/ScratchAddons#5909] [Discussion]
Prioritization helps us get important work done and less important work out of the way. However, for most issues, we can't glance and get an idea of how important they are because they don't have priority labels and they're not included in milestones or projects.

This serves as a discussion for us to talk about how we could improve and make more use out of priority labels.

#### 🏷 Specify a priority level for all issues we will be working on

This helps us focus more on what we should be doing in the moment.

#### 🗂 Add more priority levels

I think a good start would be priorities 1-5, where lower numbers are more important and higher numbers are less important, like rankings.

Depending on how we design our priority system, we might end up using more or less priority levels.

#### 📌 Separate to-dos from suggestions

We could use labels or milestones to identify issues that will be worked on.

For suggestions we are not sure about working on yet, we won't add a label or milestone, which implies that we aren't ready to work on it yet. For suggestions, bug reports, and tasks that we _will_ be working on, we will add a label or milestone, which implies that we are going to be working on it in the future.

**Comment by WorldLanguages** [comment]
> I've seen several cases now where first-time contributors don't create an issue first, which means we can't discuss their idea before they make it. To help everyone save time, I propose a new rule requiring all non-organization members to create an issue for anything they want to work on. Any pull request without a linked issue would be closed and we would explain how to create an issue and link it.

Since you wrote this, there have been 2 advancements regarding this:
1. The PR template now also says `If there aren't any [issues], please submit one first unless this is a hotfix or minor string update.` (#5889)
2. The "creating an addon" page in the docs now includes a warning at the top (https://github.com/ScratchAddons/website-v2/pull/319)

**Comment by WorldLanguages** [comment]
> We could also ask that any issue they link to have a priority label, which means we want to work on it.

I'm not sure if it would be beneficial to be that strict. I suggest to discuss this after we actually enhance the prioritization system to avoid this from slowing down other parts of the suggestion.

**Comment by WorldLanguages** [comment]
> This helps us separate suggestions from to-dos

I think this distinction is very important and we need a better way to reflect it. It doesn't necessarily need to be a label-based distinction. We could also create a different milestone for TODOs, possibly grouped by month of the year instead of version number.

**Comment by WorldLanguages** [comment]
> We would try to review pull requests tied to issues with higher priorities before lower ones, as well as pull requests in active milestones and projects.

We could attempt to do that, but I'm not sure if it could result in reviewing feeling like a robotic task. We are all volunteers so forcing a specific "priority order" for every PR could be counter-productive and lead to procrastination. It's tempting to try to borrow practices from big open source projects or software companies, but both incentives and project size are different here.

As a summary of all my 4 posts today, I think we should begin by clearly separating TODOs and bug fixes (for example #5697) from suggestions and medium/big feature additions.

**Reply by WorldLanguages** [reply]
Note that we do have priority labels, but we rarely use them. We should be using them more often, but not that often.

**Reply by WorldLanguages** [reply]
The now inactive "what's happening board" was also an attempt at grouping issues/PR by type.

**Comment by DNin01** [comment]
I'm also wondering what we should do about the descriptions. The current descriptions are as follows:
- Security bug and other release blockers
- Crashes, critical bugs and important features
- Bugs and feature additions
- Documentation, UI, and minor/not important changes

They're basically examples of what goes in each priority level, but just because a change is considered small doesn't always mean it's not important. Take the priority-4 label's description, for example. Some documentation and UI changes are definitely more important.

My description ideas at this time:
- Critical. Includes security bugs and release blockers
- High priority. Includes crashes, major bugs and important features
- Medium priority. Includes bugs and helpful features
- Low priority. Includes less important features

I also have some new colors in mind:
- `#F6090E`
- `#FDB632`
- `#F9F077`
- `#AAFF8A`

Do you like those colors or are the labels fine the way they are?

**Reply by WorldLanguages** [reply]
@DNin01 Go ahead and change them - store the old colors and descriptions somewhere in case someone complains about the change

**Reply by DNin01** [reply]
I will!

**Reply by Samq64** [reply]
Could the word helpful be replaced with useful in the priority 2 description and minor bugs be added to the priority 4 one?
